### PR TITLE
Fix missing "!=" support for WhereFilterFactory, and tighten up the char literal matching regex

### PR DESCRIPTION
**Pull request by rcaudy**

### CODE REVIEW

File: engine/table/src/main/java/io/deephaven/engine/table/impl/select/WhereFilterFactory.java [hunk 1]
```diff
@@ -56,12 +60,20 @@ public WhereFilter getExpression(String expression, Matcher matcher, Object... a
                 }
                 log.debug().append("WhereFilterFactory creating MatchFilter for expression: ").append(expression)
                         .endl();
-                return new MatchFilter(MatchFilter.CaseSensitivity.MatchCase, columnName, matcher.group(2));
+                // @formatter:off
+                return new MatchFilter(
+                        MatchFilter.CaseSensitivity.MatchCase,
+                        matcher.group(3) == null // We must have matched EQUALS_PTRN, rather than NOT_EQUALS_PTRN
```
Comment (cpwright) [inline engine/table/src/main/java/io/deephaven/engine/table/impl/select/WhereFilterFactory.java:66]: this is very subtle

**Review by cpwright — APPROVED** [review]

File: engine/table/src/main/java/io/deephaven/engine/table/impl/select/WhereFilterFactory.java (hunk 1 above)
Comment (rcaudy) [reply engine/table/src/main/java/io/deephaven/engine/table/impl/select/WhereFilterFactory.java:66]: That's why it has a comment. It's the same way as ICASE and NOT work.

**Review by cpwright — APPROVED** [review]

File: engine/table/src/main/java/io/deephaven/engine/table/impl/select/WhereFilterFactory.java (hunk 1 above)
Comment (rcaudy) [reply engine/table/src/main/java/io/deephaven/engine/table/impl/select/WhereFilterFactory.java]: I re-did the whole class' handling of matcher groups, see if this is better.

**Review by cpwright — APPROVED** [review]

**Review by cpwright — APPROVED** [review]

**Comment by rcaudy** [comment]
Fixes #1818

**Review by cpwright — APPROVED** [review]

**Pull request merged**